packages/react-meteor-data/ReactMeteorData.jsx

```diff
@@ -166,4 +122,61 @@ class MeteorDataManager {
   }
 }
 
-export default ReactMeteorData;
+export const ReactMeteorData = {
+  componentWillMount() {
+    this.data = {};
+    this._meteorDataManager = new MeteorDataManager(this);
+    const newData = this._meteorDataManager.calculateData();
+    this._meteorDataManager.updateData(newData);
+  },
+
+  componentWillUpdate(nextProps, nextState) {
+    const saveProps = this.props;
+    const saveState = this.state;
+    let newData;
+    try {
+      // Temporarily assign this.state and this.props,
+      // so that they are seen by getMeteorData!
+      // This is a simulation of how the proposed Observe API
+      // for React will work, which calls observe() after
+      // componentWillUpdate and after props and state are
+      // updated, but before render() is called.
+      // See https://github.com/facebook/react/issues/3398.
+      this.props = nextProps;
+      this.state = nextState;
+      newData = this._meteorDataManager.calculateData();
+    } finally {
+      this.props = saveProps;
+      this.state = saveState;
+    }
+
+    this._meteorDataManager.updateData(newData);
+  },
+
+  componentWillUnmount() {
+    this._meteorDataManager.dispose();
+  },
+};
+
+export function connect({ getMeteorData, pure = true }) {
+  const BaseComponent = pure ? React.PureComponent : React.Component;
+  return (WrappedComponent) => {
+    class ReactMeteorDataComponent extends BaseComponent {
+      getMeteorData() {
+        return getMeteorData(this.props);
+      }
+      render() {
+        return <WrappedComponent {...this.props} {...this.data} />;
+      }
+    }
+
+    ReactMeteorDataComponent.prototype.componentWillMount =
```

Comment:
Instead of assigning in this way, couldn't these `componentWill{Mount,Update,Unmount}` methods represent an intermediary object which was created (first) by `extend`ing onto `BaseComponent` and then have `getMeteorData` and `render` further `extend`ed onto that intermediary object to get the resulting `ReactMeteorDataComponent`?